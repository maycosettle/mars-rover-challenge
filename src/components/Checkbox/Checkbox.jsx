import useStyles from './styles';

const Checkbox = ({ label, description, checked, onChange }) => {
  const styles = useStyles();

  return (
    <div className={styles.container}>
      <div className={styles.inputContainer}>
        <input
          id="comments"
          aria-describedby="comments-description"
          name="comments"
          type="checkbox"
          checked={checked}
          onChange={onChange}
          className={styles.input}
        />
      </div>
      <div className={styles.labelContainer}>
        <label htmlFor="comments" className={styles.label}>
          {label}
        </label>
        <span id="comments-description" className={styles.span}>
          {description}
        </span>
      </div>
    </div>
  );
};

export default Checkbox;
