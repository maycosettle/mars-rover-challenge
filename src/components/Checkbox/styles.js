const useStyles = () => ({
  container: 'relative flex items-start',
  inputContainer: 'flex items-center h-5',
  input: 'focus:ring-primary-500 h-4 w-4 text-primary-600 border-gray-300 rounded',
  labelContainer: 'ml-3 text-sm',
  label: 'font-medium text-gray-700',
  span: 'text-gray-500 ml-1',
});

export default useStyles;
