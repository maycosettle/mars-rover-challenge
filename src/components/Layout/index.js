export * from './PageLayout';
export * from './PaperLayout';
export * from './PhotoLayout';
