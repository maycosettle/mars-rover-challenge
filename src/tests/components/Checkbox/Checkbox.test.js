import { shallow } from 'enzyme';
import { Checkbox } from 'components/Checkbox';

describe('tests on <Checkbox />', () => {
  const wrapper = shallow(<Checkbox />);

  test('Should render Checkbox correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
