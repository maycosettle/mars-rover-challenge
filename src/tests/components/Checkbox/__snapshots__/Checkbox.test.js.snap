// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <Checkbox /> Should render Checkbox correctly 1`] = `
<div
  className="relative flex items-start"
>
  <div
    className="flex items-center h-5"
  >
    <input
      aria-describedby="comments-description"
      className="focus:ring-primary-500 h-4 w-4 text-primary-600 border-gray-300 rounded"
      id="comments"
      name="comments"
      type="checkbox"
    />
  </div>
  <div
    className="ml-3 text-sm"
  >
    <label
      className="font-medium text-gray-700"
      htmlFor="comments"
    />
    <span
      className="text-gray-500 ml-1"
      id="comments-description"
    />
  </div>
</div>
`;
